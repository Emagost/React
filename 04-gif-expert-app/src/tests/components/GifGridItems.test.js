import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';


describe('Test GifGridItme', () => {
    
    const title = 'Title';
    const url = 'http://localhost/algo.jpg'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test('should show corretly the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should return a paragraph with the text', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('should return an img = props and alt = props', () => {
        
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    })

    test('should have animate__fadeIn', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true); 

        //const className = div.prop('className');
        //expect(className.includes('animate__fadeIn')).toBe(true);
        
    })
    
    

})
