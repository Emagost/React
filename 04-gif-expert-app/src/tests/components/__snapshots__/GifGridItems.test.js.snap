// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItme should show corretly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title"
    src="http://localhost/algo.jpg"
  />
  <p>
     
    Title
     
  </p>
</div>
`;
